.circleci/scripts/test-run-e2e-timeout-minutes.ts

```diff
@@ -0,0 +1,8 @@
+import { filterE2eChangedFiles } from '../../test/e2e/changedFilesUtil';
+
+const changedOrNewTests = filterE2eChangedFiles();
+
+//15 minutes, plus 3 minutes for every changed file, up to a maximum of 30 minutes
+const extraTime = Math.min(15 + changedOrNewTests.length * 3, 30);
+
+console.log(extraTime);
```

Comment:
_Ackchyually_ this isn't "extra time"; it's a timeout time _plus_ extra time. :-)

Not a show stopper for the PR, by any means. This PR is two changes: reduce the default timeout time from 20 minutes to 15 minutes, and add 3 minutes of extra time for each changed e2e file.


Reducing the time from 20 to 15 is what I'm curious out. How did you arrive at 15?

Reply:
@HowardBraham did the implementation and might know why

Reply:
cool. I'm 👍  on giving it a go with 15 minutes